contracts/InterestRate/StableRate.sol

```diff
@@ -0,0 +1,248 @@
+     */
+    function utilizationRate(
+        uint256 cash,
+        uint256 borrows,
+        uint256 reserves
+    ) public pure returns (uint256) {
+        // Utilization rate is 0 when there are no borrows
+        if (borrows == 0) {
+            return 0;
+        }
+
+        return (borrows * BASE) / (cash + borrows - reserves);
+    }
+
+    /**
+     * @notice Calculates the stable loan rate of the market: `stableborrows / totalborrows`
+     * @param stableborrows The amount of stable borrows in the market
+     * @param totalborrows The amount of total borrows in the market
+     * @return The stable loan rate as a mantissa between [0, BASE]
+     */
+    function stableLoanRate(uint256 stableborrows, uint256 totalborrows) public pure returns (uint256) {
+        // Loan rate is 0 when there are no stable borrows
+        if (totalborrows == 0) {
+            return 0;
+        }
+
+        return (stableborrows * BASE) / totalborrows;
+    }
+
+    /**
+     * @notice Calculates the stable loan rate of the market: `stableborrows / totalborrows`
+     * @param loanRate Loan rate for stable borrows in the market
+     * @return The difference between the loan rate and optimal loan rate as a mantissa between [0, BASE]
+     */
+    function calculateLoanRateDiff(uint256 loanRate) internal view returns (uint256) {
+        if (loanRate == 0) {
+            return 0;
+        }
+
+        if (optimalStableLoanRate > loanRate) {
+            return 0;
+        }
+
+        return ((loanRate - optimalStableLoanRate) * BASE) / (BASE - optimalStableLoanRate);
+    }
+
+    function getBorrowRate(
+        uint256 cash,
+        uint256 stableborrows,
+        uint256 totalborrows,
+        uint256 reserves
+    ) external view returns (uint256) {
+        uint256 util = utilizationRate(cash, totalborrows, reserves);
+        uint256 loanRate = stableLoanRate(stableborrows, totalborrows);
+        uint256 excessLoanRate = calculateLoanRateDiff(loanRate);
+
+        if (util <= kink) {
+            return
+                ((util * multiplierPerBlock) / BASE) +
+                baseRatePerBlockForStable +
+                ((util * multiplierPerBlockForStable) / BASE) +
+                ((stableRatePremium * excessLoanRate) / BASE);
+        } else {
+            uint256 excessUtil = util - kink;
+            return
+                ((kink * multiplierPerBlock) / BASE) +
+                baseRatePerBlockForStable +
+                ((kink * multiplierPerBlockForStable) / BASE) +
+                ((excessUtil * jumpMultiplierPerBlockForStable) / BASE) +
+                ((stableRatePremium * excessLoanRate) / BASE);
+        }
+    }
+
+    /**
+     * @notice Internal function to update the parameters of the interest rate model
+     * @param _baseRatePerBlockForStable The approximate target base APR, as a mantissa (scaled by BASE)
+     * @param _multiplierPerBlock The rate of increase in interest rate wrt utilization (scaled by BASE)
```

Comment:
nit: should we write out `with respect to`. to save space we could also write `per`